ambry-store/src/test/java/com.github.ambry.store/CompactionDetailsTest.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.UtilsTest;
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests for {@link CompactionDetails}.
+ */
+public class CompactionDetailsTest {
+
+  /**
+   * Tests the serialization and deserialization of {@link CompactionDetails} and verifies eqaulity of the original and
+   * deserialized forms.
+   * @throws IOException
+   */
+  @Test
+  public void serDeTest() throws IOException {
+    int segmentCount = TestUtils.RANDOM.nextInt(10) + 1;
+    List<String> segmentsUnderCompaction = new ArrayList<>();
+    for (int i = 0; i < segmentCount; i++) {
+      int stringSize = TestUtils.RANDOM.nextInt(10) + 1;
+      segmentsUnderCompaction.add(UtilsTest.getRandomString(stringSize));
+    }
+    long referenceTime = SystemTime.getInstance().milliseconds();
+    int swapSpaceCount = TestUtils.RANDOM.nextInt(10) + 1;
+
+    // without extra segment
+    CompactionDetails details = new CompactionDetails(referenceTime, segmentsUnderCompaction, null, swapSpaceCount);
```

Comment:
Removed the javadoc that says it can be empty. Expect it to be `null` if no extra segment is required.